### ISSUE TITLE
Add readability through OOP and basic documentation

### ISSUE THREAD

**Issue by Moosems** [Futura-Py/notes#76]

**Comment by not-nef** [comment]
no oop.

**Comment by im-coder-lg** [comment]
@not-nef OOP gives more clarity. Last time I checked, many people used OOP to have more neatness and control over Python. We can help you with this, you know. I don't use much OOP, but I can promise to test the code logics that uses OOP code here.

**Comment by not-nef** [comment]
im against oop because i dont understand it AT ALL. So if we switch to oop with all the classes and shit i wont be able to maintain this project no more

**Comment by sumeshir26** [comment]
Well its your wish but in the long run It is easier to maintain with classes since everything is seperate

**Comment by Moosems** [comment]
> im against oop because i dont understand it AT ALL. So if we switch to oop with all the classes and shit i wont be able to maintain this project no more

Then learn it. You don't learn by not doing. You learn by failing and changing

**Comment by Moosems** [comment]
> Well its your wish but in the long run It is easier to maintain with classes since everything is seperate

Plus it keeps everything self contained and makes tracking down problems easier

**Comment by not-nef** [comment]
yeah ok but ill leave the transition to you because you seem to understand it better

**Comment by Moosems** [comment]
Ok 👍. I can document it if that helps

**Comment by Moosems** [comment]
not-nef I actually used to code just like you do but I promise that if you learn oop it will change your world

**Comment by not-nef** [comment]
yeah but for now im gonna focus on some other projects

**Comment by im-coder-lg** [comment]
Guys I have some ideas in mind. Like a weather app using Gtk4+ and PyQt or something for Linux and Tukaan for Windows and Mac. Anyone interested? We will be using a lot of APIs, mainly from OpenWeatherMap.

**Comment by im-coder-lg** [comment]
Maybe there we could use OOP. And scratch GTK4+ and all that, we'll just use Tukaan. Or mainstream Tkinter with sv-ttk.

**Comment by Moosems** [comment]
Normally it's better to stick with one GUI ToolKit however I am more than interested in this weather app

**Comment by sumeshir26** [comment]
@im-coder-lg Im out cuz i got no time for coding, and all my coding time these days is spent on Jou... The app seems interesting tho...

**Comment by not-nef** [comment]
@im-coder-lg definetely in if we use tkinter

**Comment by im-coder-lg** [comment]
Okay @sumeshir26 I get it. We will do it during May though, to get the most time.

**Comment by Moosems** [comment]
> @im-coder-lg Im out cuz i got no time for coding, and all my coding time these days is spent on Jou... The app seems interesting tho...

Remember that Jou is still in early development and subject to big changes. It also isn't very strong in any categories yet

**Comment by not-nef** [comment]
better learn c tbh

**Comment by Moosems** [comment]
> Okay guys, and @sumeshir26 I get it. We will do it during May though, to get the most time.

You want to create the repo to start? Also can we change the file structure from a src type folder to a more open one? I can set up a pyinstaller build action so we can test our app upon push

**Comment by Moosems** [comment]
Not-nef you know c?

**Comment by not-nef** [comment]
uuuuhm... sure, but python is my strong suit

**Comment by Moosems** [comment]
Do we want to try a messaging thing again? IRC is pretty good for development

**Comment by not-nef** [comment]
did i miss something? what messenger

**Comment by Moosems** [comment]
I remember a while back that ee tried finding a messaging service so we could talk faster

**Comment by not-nef** [comment]
whats wrong about github, its not like were in a rush

**Comment by Moosems** [comment]
Well it's easier for debugging when you can quickly respond and chat long amounts that others don't need to read through to help

**Comment by Moosems** [comment]
Me and Akuli chat using irc.libera.chat on ##learnpython (Try his mantaray client)

**Comment by sumeshir26** [comment]
@Futura-Py/reviewers Shall I create the repo? Need a name

**Comment by not-nef** [comment]
*weather* (Futura Weather)

**Comment by Moosems** [comment]
I guess I'm part of Futura now?

**Comment by sumeshir26** [comment]
OKay

**Comment by not-nef** [comment]
> I guess I'm part of Futura now?

sure

**Comment by Moosems** [comment]
Nice 👍

**Comment by sumeshir26** [comment]
So what framework should we use? Tukaan, Tkinter, GTK+, Something of C, or what?

**Comment by Moosems** [comment]
Tkinter? It's great for a prototype but not great once it gets really nice

**Comment by Moosems** [comment]
What exactly is the plan/design for it? @im-coder-lg This was your plan so maybe you choose?

**Comment by not-nef** [comment]
The best framework ever is pyqt. Its hard to learn but if you know it you can basically write fluent apps

**Comment by Moosems** [comment]
You learned it Nef? I've wanted to get into it for a long time but never cared to

**Comment by not-nef** [comment]
no. not yet. but i desperately want to. just need to get my ass up

**Comment by Moosems** [comment]
Let's do it then

**Comment by sumeshir26** [comment]
You know what, I'm working on this as well. Lemme push everything else aside.

**Comment by sumeshir26** [comment]
I guess time to watch a PyQt5 tutorial...
And do you guys wanna use PySide6 or PyQt6

**Comment by Moosems** [comment]
PyQt6 personally

**Comment by im-coder-lg** [comment]
Wait, can you theme apps when you use PyQt?

**Comment by not-nef** [comment]
afaik yeah

**Comment by Moosems** [comment]
> Wait, can you theme apps when you use PyQt?

With how popular it is I'd be surprised if you can't

**Comment by sumeshir26** [comment]
@im-coder-lg Qt follows the system theme regardless of the system :)

**Comment by sumeshir26** [comment]
And I found this if we want to have Sun Valley Theming: https://github.com/zhiyiYo/PyQt-Fluent-Widgets

**Comment by not-nef** [comment]
YES THATS WHAT IVE BEEN LOOKING FOR TYSM

**Comment by im-coder-lg** [comment]
![yeah!](https://media.tenor.com/iN5PHnKGu30AAAAM/high-five-the-flash.gif)

# LESGOO
